sklearn/feature_selection/tests/test_base.py

```diff
@@ -16,13 +15,13 @@ def __init__(self, step=2):
         self.step = step
 
     def fit(self, X, y=None):
-        X = check_array(X, accept_sparse="csc")
-        self.n_input_feats = X.shape[1]
+        X = self._validate_data(X, accept_sparse="csc")
         return self
 
     def _get_support_mask(self):
-        mask = np.zeros(self.n_input_feats, dtype=bool)
-        mask[:: self.step] = True
+        mask = np.zeros(self.n_features_in_, dtype=bool)
+        if self.step >= 1:
+            mask[:: self.step] = True
```

Comment:
is this a bug you're fixing here?

Reply:
This is a new feature to `StepSelector`, where `step < 1` means the mask is all `False` and no features are selected. Note that `StepSelector` is only used for testing.

I updated the docstring to mention this fact: [`1e5b8ee` (#25102)](https://github.com/scikit-learn/scikit-learn/pull/25102/commits/1e5b8eeb00dee61f5494bf1e7626c834165c4acd)